lib/avatar.cpp

```diff
@@ -178,6 +191,7 @@ bool Avatar::updateUrl(const QUrl& newUrl)
         return false;
 
     d->_url = newUrl;
+    d->_localFile = QUrl::fromLocalFile(QStandardPaths::writableLocation(QStandardPaths::CacheLocation) + "/" + newUrl.authority() + "_" + newUrl.fileName() + ".png");
```

Comment:
Maybe make a separate method in `Private` to assign all these, to avoid code duplication. Or even go wild and recreate the entire `d` object, making sure to abandon `_thumbnailRequest` before it.